packages/styles/README.md

```diff
@@ -0,0 +1,264 @@
+# Styling lib
+
+Fully fledged library to provide best DX for writing styles. In core it uses library `fela` which seems like a good choice for quality, simplicity, performance. I strongly recommend reading [Fela docs](https://fela.js.org/docs/latest/intro/getting-started), especially sections [Motivation](https://fela.js.org/docs/latest/intro/motivation) and [Principles](https://fela.js.org/docs/latest/intro/principles). Big thanks to [wafflepie](https://github.com/wafflepie) for birth of this lib and inspiration.
```

Comment:
I like it, but I still have my doubts if the benefits from this library outweighs the costs for the maintenance. Especially if we use it just for suite-native for now. There is a lot of code relevant only for usage on web. I see the biggest potential of this library for the web (all the benefits mentioned in [https://fela.js.org/docs/latest/intro/benefits](https://fela.js.org/docs/latest/intro/benefits) seem to me relevant only for the web). The only benefit I see for the React Native usage is nice way to write dynamic styles.

I'd love to hear @dahaca's opinion. Could you take a look at this styling package, please?

Reply:
I think it maintenance cost won't be any high if any, in my previous job we did not touched this for like 1,5 year and I do not expect any changes that would be necessary in future, except adding some more features if we will want to.

I think we really want ability to write easily dynamic styles in RN and I know it can be done without fela, but I think writing some custom solution (it actually was one of previous version of this lib) is much worse for mainteneance than this solution. In my experience is better to use some 3rd party lib to have some good quality foundations, because fela is doing some caching for RN which is better to "outsource" and also it has some foundations to writing some plugins/enhancers and this kind of stuff. For example I plan to write some plugin to support media queries also on React Native platform, which will be useful if we want to support tablets etc.

About the web support even when it is not used now, I don't see this as big deal because it won't be included in any build and all fela dependencies are from one monorepo so we can count then (for security purposes) as one package. If we will decide to not use this for new v2 components for web, code related to web could be deleted, which will be much easier than delete it now and maybe add later. Also deleting it now will also discourage people to try it for web and I still strongly believe if others will give it a try, they will like.

Reply:
@dahaca As I said a definitely not going to force anyone to use it for web but I still think you may start to like it when you will start to use it for native part.

If won't like and it you will decide to not use it, it will be really simple to remove web part later and it think it won't make any harm if it will be there for a while.

Reply:
About the performance argument (I going to shot myself into leg) but I would definitely not to use that as argument why choose Fela and I think you can totally ignore that for that decision. 

Probably main point of that was try to show is how ridiculously slow `styled-components` could be if I compare them to anything else. I know these benchmarks are not meant to be like super precise but difference is soo huge, even if it will be like 20x faster it still will be slower than anything else.

As you said it's "good enough" right now, but these `ms` could be difference between 60fps experience and maybe little bit experience especially on some slower computers.

But you definitely don't need `fela` to make it faster, you can use for example `emotion` which is using same syntax as `styled-components` and you can nearly plug&play replace it, but it's much faster and better built IMHO. So that's reason why you can totally ignore performance as argument for Fela.

Reply:
Agree with everything. And I can totally see how it benefits Native + we need all the FPS for those 120hz screens 😂